### PR TITLE
Update maintainers.md

### LINKED ISSUE #735: Update the Maintainers file

### PR DESCRIPTION
**Pull request by dharmjit**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** :
Fixes #735 

**Additional information**

**Special notes for your reviewer**

### CODE REVIEW

File: MAINTAINERS.md
Comment (anusha94) [inline MAINTAINERS.md:7]: Sort names alphabetically? That was the convention that we followed earlier.

File: MAINTAINERS.md
Comment (anusha94) [inline MAINTAINERS.md:7]: Time to create a new list for Emeritus maintainers :)
Refer https://github.com/vmware-tanzu/velero/blob/main/MAINTAINERS.md#emeritus-maintainers

File: MAINTAINERS.md
Comment (anusha94) [inline MAINTAINERS.md:19]: I'd rather have my name removed from the engineering list as I won't be able to actively contribute anymore.

File: MAINTAINERS.md
Comment (anusha94) [inline MAINTAINERS.md:24]: I am not sure of having an emeritus contributors list. But I think we can have one. BYOH wouldn't be what it is without the initial contributions of Hui, Yixing, Wuming, Rousko, and others :) 
I'd leave it to you though.

Comment (dharmjit) [reply MAINTAINERS.md:24]: I thought about this and it definitely makes sense to keep all the core contributors to date. I will take some hints from other OSS projects and any guidance I can get. Keeping a emeritus contributors list also seems like an option.

File: MAINTAINERS.md
Comment (dharmjit) [reply MAINTAINERS.md:7]: Thanks for suggesting this.

**Comment by shivi28** [comment]
Thanks @dharmjit for these changes. 😄 
Special thanks to @anusha94 for all her contributions and guidance 🎉  🙏

**Review by shivi28 — APPROVED** [review]

**Review by Madhur97 — APPROVED** [review]

**Pull request merged**
